core/UrlHelper.php

```diff
@@ -135,10 +143,20 @@ public static function isLookLikeUrl($url)
     {
         return preg_match('~^(([[:alpha:]][[:alnum:]+.-]*)?:)?//(.*)$~D', $url, $matches) !== 0
             && strlen($matches[3]) > 0
-            && !in_array($matches[1], array('javascript:', 'vbscript:', 'data:'))
+            && !preg_match('/^(javascript:|vbscript:|data:)/i', $matches[1])
             ;
     }
 
+    public static function isLookLikeSafeUrl($url)
+    {
+        if (strpos($url, ':') === false) {
+            return true;
+        }
+
+        $protocol = explode(':', $url, 2)[0];
+        return preg_match('/^' . implode('|', self::$validLinkProtocols) . '/i', $protocol);
```

Comment:
small tweak maybe... should the protocol also check for ends with this? as in `'/^' . implode('|', $validLinkProtocols) . '$/i'` otherwise `smstest` might be considered a valid protocol?